### PR TITLE
Programming auto grader fixes

### PR DESCRIPTION
**Pull request by lowjoel**
- Grade the student's actual answer
 - Fix computation of awarded grade

### CODE REVIEW

File: app/services/course/assessment/answer/programming_auto_grading_service.rb [hunk 1]
```diff
@@ -42,6 +42,7 @@ def build_submission_files(answer)
   # @param [Course::Assessment::ProgrammingPackage] package The package to import.
   # @return [Course::Assessment::ProgrammingEvaluationService::Result]
   def evaluate_package(question, package)
+    package.save
```
Comment (allenwq) [inline app/services/course/assessment/answer/programming_auto_grading_service.rb]: in what case files inside package are not committed ?

Comment (lowjoel) [inline app/services/course/assessment/answer/programming_auto_grading_service.rb]: when the template is merged with the student's answers.

Comment (allenwq) [inline app/services/course/assessment/answer/programming_auto_grading_service.rb]: Okay, but I think the logic should be handled outside, it's out of the duty of evaluate package, what do you think ?

Comment (lowjoel) [inline app/services/course/assessment/answer/programming_auto_grading_service.rb]: complete agree, fixed.

**Comment by allenwq** [comment]
Thanks !

**Pull request merged**